path_roles.go

```diff
@@ -345,13 +345,18 @@ func (b *azureSecretBackend) createPersistedApp(ctx context.Context, req *logica
 		return err
 	}
 
+	assignmentIDs, err := c.generateAssignmentIDsFromRole(role)
+	if err != nil {
+		return err
+	}
```

Comment:
Might be a good idea to wrap the error here.

Example:
```return fmt.Errorf("failed to generate assignmentIDs for PersistedApp roles: %w", err)```

Not sure if my example error text is accurate wording but hopefully it makes enough sense to convey my intention.

Reply:
Updated in latest commit